rbac/management/group/view.py

```diff
@@ -1126,14 +1141,41 @@ def remove_service_accounts(
 
             raise ValueError(f"Service account(s) {service_account_ids_diff} not found in the group '{group.name}'")
 
+        removed_service_accounts = []
         # Remove service accounts from the group.
         with transaction.atomic():
             for service_account in valid_service_accounts:
                 group.principals.remove(service_account)
+                removed_service_accounts.append(service_account)
 
         logger.info(
             f"[Request_id:{request_id}] {valid_service_account_ids} "
             f"removed from group {group.name} for org id {org_id}."
         )
         for username in service_accounts:
             group_principal_change_notification_handler(self.request.user, group, username, "removed")
```

Comment:
One note around the notification handlers: previously these happend _after_ the writes committed. So the semantics you had prior to this were that you might not send a message (due to failure) even if you successfully committed something.

With this change, that's different. Now it's the other way around: because we haven't committed yet, we send the message regardless of whether our commit rolls back, which means you might instead send a message when you shouldn't have.

So it's whether you want to err on the side of sending when you shouldn't, vs not sending when you should've. I'm not sure which is better–maybe even this is an improvement–but just something to keep in mind for edge cases.

Reply:
This is good point to consider - in addition if sending into kafka will fail - I guess it will rollback whole transaction. Which is probably fine as it means that debezium will also not work :) 

(I will keep as it is - as I see role dual write has same change   )

Reply:
That's a good point. It means if Kafka is down, more functionality is impacted (severely!). I suppose that is another reason to consider moving notifications to use the outbox pattern, too. Not only would it be atomic, but it would also be more available (and probably reduce latencies on API calls, too).

We may want to reconsider the role changes that brought that into the TX as well, but I'll open an issue for it and we can discuss with the team.